src/arkode/arkode_arkstep_io.c

```diff
@@ -1618,17 +1659,19 @@ int ARKStepGetTimestepperStats(void *arkode_mem, long int *expsteps,
                                  &ark_mem, &step_mem);
   if (retval != ARK_SUCCESS)  return(retval);
 
-  /* set expsteps and accsteps from adaptivity structure */
-  *expsteps = ark_mem->hadapt_mem->nst_exp;
-  *accsteps = ark_mem->hadapt_mem->nst_acc;
-
-  /* set remaining outputs */
+  /* set integrator outputs */
   *step_attempts = ark_mem->nst_attempts;
   *fe_evals      = step_mem->nfe;
   *fi_evals      = step_mem->nfi;
   *nlinsetups    = step_mem->nsetups;
   *netfails      = ark_mem->netf;
 
+  /* get expsteps and accsteps from heuristics object */
+  retval = SUNHeuristicsGetNumExpSteps(ark_mem->hconstraints, expsteps);
+  if (retval != ARK_SUCCESS) { return(ARK_HEURISTICS_ERR); }
+  retval = SUNHeuristicsGetNumAccSteps(ark_mem->hconstraints, accsteps);
+  if (retval != ARK_SUCCESS) { return(ARK_HEURISTICS_ERR); }
```

Comment:
Are these counters retained across are reset/resize calls and set to zero on re-initialization? If the controller is replaced during a run are these counts lost?

Reply:
These are wiped out whenever the heuristics object is `Reset` (this is called automatically upon heuristics creation).
`arkInit` will only call `SUNTimestepHeuristics_Reset` when it is called with `init_type == FIRST_INIT`.